Kudu.Contracts/Settings/DiagnosticsSettings.cs

```diff
@@ -114,5 +124,72 @@ public bool RemoveSetting(string key)
         {
             return _settings.Union(_extraSettings).GetEnumerator();
         }
+
+        public void EnsureValid()
+        {
+            string[] itemKeys = new string[] { AzureDriveTraceLevelKey, AzureTableTraceLevelKey, AzureBlobTraceLevelKey };
+            foreach (var key in itemKeys)
+            {
+                object value = this.GetSetting(key);
+                if (!Enum.IsDefined(typeof(TraceEventType), value))
+                {
+                    throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, "Invalid Key: '{0}', Value: '{1}'", key, value));
+                }
+            }
+        }
+
+        public static bool AutoCorrect(ITracer tracer, string rawContent, out string newRawContent)
```

Comment:
This method seems really complicated, and requires maintenance to remember to add new settings to a static list and to add new codeblocks to this method.